op-chain-ops/genesis/setters_test.go

```diff
@@ -0,0 +1,50 @@
+package genesis
+
+import (
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestWipePredeployStorage(t *testing.T) {
+	rawDB := rawdb.NewMemoryDatabase()
+	rawStateDB := state.NewDatabaseWithConfig(rawDB, &trie.Config{
+		Preimages: true,
+		Cache:     1024,
+	})
+	stateDB, err := state.New(common.Hash{}, rawStateDB, nil)
+	require.NoError(t, err)
+
+	storeVal := common.Hash{31: 0xff}
+
+	for _, addr := range predeploys.Predeploys {
+		a := *addr
+		stateDB.SetState(a, storeVal, storeVal)
+		stateDB.SetBalance(a, big.NewInt(99))
+		stateDB.SetNonce(a, 99)
+	}
+
+	root, err := stateDB.Commit(false)
+	require.NoError(t, err)
+
+	err = stateDB.Database().TrieDB().Commit(root, true)
+	require.NoError(t, err)
+
+	require.NoError(t, WipePredeployStorage(stateDB))
+
+	for _, addr := range predeploys.Predeploys {
+		a := *addr
+		if FrozenStoragePredeploys[a] {
+			require.Equal(t, storeVal, stateDB.GetState(a, storeVal))
+		} else {
+			require.Equal(t, common.Hash{}, stateDB.GetState(a, storeVal))
+		}
+		require.Equal(t, big.NewInt(99), stateDB.GetBalance(a))
```

Comment:
Wait maybe i'm reading this wrong, but shouldn't this be checking that the balance is 0? The balance isn't wiped?

Reply:
We're checking that the balance is `99`, which is what we set every predeploy's balance to in the first for loop above. We need to make sure that the balance and nonce are properly carried over.

Reply:
The balance is not wiped by `CreateAccount`, it gets carried over - this is what led to the Sherlock doubling issue.